pystac_monty/sources/ibtracs.py

```diff
@@ -52,23 +52,45 @@ def _parse_csv(self) -> List[Dict[str, str]]:
             csv_data.append(row)
         return csv_data
 
+    def get_data(self):
+        """Yield storm data grouped by SID from a sorted CSV."""
+
+        with open(self.data, mode="r", newline="", encoding="utf-8") as f:
+            reader = csv.DictReader(f)
+            current_storm_id = None
+            storm_data = []
+
+            for row in reader:
+                storm_id = row["SID"]
+                if storm_id != current_storm_id:
+                    if storm_data:
+                        yield storm_data
+                    storm_data = [row]
+                    current_storm_id = storm_id
+                else:
+                    storm_data.append(row)
+            if storm_data:
+                yield storm_data
+
 
 class IBTrACSTransformer(MontyDataTransformer[IBTrACSDataSource]):
     """Transforms IBTrACS tropical cyclone data into STAC Items."""
 
     hazard_profiles = MontyHazardProfiles()
-    source_name = 'ibtracs'
+    source_name = "ibtracs"
 
     def get_stac_items(self) -> typing.Generator[Item, None, None]:
         self.transform_summary.mark_as_started()
         # TODO: Use sax xml parser for memory efficient usage
-        csv_data = self.data_source._parse_csv()
-        csv_data.sort(key=lambda x: x.get("SID", " "))
-        for storm_id, storm_data_iterator in itertools.groupby(csv_data, key=lambda x: x.get("SID", " ")):
-            storm_data = list(storm_data_iterator)
-            self.transform_summary.increment_rows(len(storm_data))
+        csv_data = self.data_source.get_data()
+        for storm_data in csv_data:
+            if storm_data[0]["SID"] == " ":
```

Comment:
use strip method instead

Reply:
Also make sure to have a check if the storm_data[0]["SID"] is a string and not None type to apply strip()